staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3906,9 +3906,18 @@ type ServiceSpec struct {
 	// of peer discovery.
 	// +optional
 	PublishNotReadyAddresses bool `json:"publishNotReadyAddresses,omitempty" protobuf:"varint,13,opt,name=publishNotReadyAddresses"`
+
 	// sessionAffinityConfig contains the configurations of session affinity.
 	// +optional
 	SessionAffinityConfig *SessionAffinityConfig `json:"sessionAffinityConfig,omitempty" protobuf:"bytes,14,opt,name=sessionAffinityConfig"`
+
+	// topologyKeys is a preference-order list of topology keys which clients should respect when accessing this Service.
+	// If any ready backends exist for index [0], they should always be chosen; only if no backends exist for index [0] should backends for index [1] be considered.
+	// If this field is specified, but all entries have no backends that match the topology of the client, the service has no backends for that client and connections should fail.
+	// The special value "" may be used to means "any node". This catch-all value, if used, only makes sense as the last value in the list.
```

Comment:
1. Might make a lot of sense to have the same example in KEP here. 
2- `If this field is specified, but all entries have no backends ...` should be `If this field is specified *and* all entries have no backends`
3. The comment is written as if it is a bullet point (caps on each line). Usually comments are written as a paragraph.

Reply:
done